src/de.hpi.swa.trufflesqueak/src/de/hpi/swa/trufflesqueak/nodes/AboutToReturnNode.java

```diff
@@ -76,8 +76,8 @@ protected final void doAboutToReturnVirtualizedNothing(final VirtualFrame frame,
         @Specialization(guards = {"hasModifiedSender(frame)"})
         protected static final void doAboutToReturn(final VirtualFrame frame, final NonLocalReturn nlr,
                         @Cached("createAboutToReturnSend()") final Dispatch2Node sendAboutToReturnNode) {
-            assert nlr.getTargetContextOrMarker() instanceof ContextObject;
-            sendAboutToReturnNode.execute(frame, FrameAccess.getContext(frame), nlr.getReturnValue(), nlr.getTargetContextOrMarker());
+            /* assert nlr.getTargetContextOrMarker() instanceof ContextObject; */
```

Comment:
If this no longer holds true, should we remove it?

Reply:
Yes!